sklearn/datasets/_samples_generator.py

```diff
@@ -306,16 +329,55 @@ def make_classification(
         scale = 1 + 100 * generator.uniform(size=n_features)
     X *= scale
 
+    indices = np.arange(n_features)
     if shuffle:
         # Randomly permute samples
         X, y = util_shuffle(X, y, random_state=generator)
 
         # Randomly permute features
-        indices = np.arange(n_features)
         generator.shuffle(indices)
         X[:, :] = X[:, indices]
 
-    return X, y
+    # feat_desc describes features in X
+    feat_desc = ["random"] * n_features
+    for i, index in enumerate(indices):
+        if index < n_informative:
+            feat_desc[i] = "informative"
+        elif index >= n_informative and index < n_informative + n_redundant:
+            feat_desc[i] = "redundant"
+        elif index >= n and index < n + n_repeated:
+            feat_desc[i] = "repeated"
+
+    if return_X_y:
+        return X, y
```

Comment:
I think this can be moved above the `feat_desc` part because if we want to simply return `X` and `y` we don't really need to first set the `feat_desc`.